### PR TITLE
Make AsciiOutputModule to print out alias branches, and extend EventContentAnalyzer to optionally print provenance information

### PR DESCRIPTION
**Pull request by makortel**
This PR includes two related changes for investigating in detail the provenance information of EDAlias products:

### AsciiOutputModule

`Event::getAllProvenance()` ignores alias branches, so `AsciiOutputModule` ignores alias branches as well. In order to print those, I 
* changed the approach to get the list of branches from `ConstProductRegistry` instead
* added a printout whether the branch is an alias for some other branch
* optionally (`verbosity > 2`) print out the PSet of the module producing the branch (and in case of alias, of the module alias is for)

### EventContentAnalyzer

I added an option (`listProvenance`) to print out (event by event) provenance information for the branches.

Tested in CMSSW_10_4_0_pre2, no changes expected.

@Dr15Jones

### CODE REVIEW

File: FWCore/Modules/src/EventContentAnalyzer.cc [hunk 1]
```diff
@@ -472,6 +509,8 @@ namespace edm {
      np = desc.addOptionalUntracked<bool>("listContent", true);
      np->setComment("If true then print a list of all the event content.");
 
+     np = desc.addOptionalUntracked<bool>("listProvenance", false);
```
Comment (makortel) [inline FWCore/Modules/src/EventContentAnalyzer.cc]: @Dr15Jones Is there a good reason for these to be optional, given that all of them have default values? Now the defaults are specified twice.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

cross-checked in a simple standalone test

**Pull request merged**